pkg/prometheus/promcfg_test.go

```diff
@@ -4314,21 +4314,29 @@ func TestKeepDroppedTargets(t *testing.T) {
 
 func TestBodySizeLimits(t *testing.T) {
 	for _, tc := range []struct {
-		version               string
-		enforcedBodySizeLimit monitoringv1.ByteSize
-		expectedErr           error
-		golden                string
+		version                     string
+		enforcedBodySizeLimit       monitoringv1.ByteSize
+		serviceMonitorBodySizeLimit monitoringv1.ByteSize
```

Comment:
(nit) better to use the same type as the CRD.
```suggestion
		serviceMonitorBodySizeLimit *monitoringv1.ByteSize
```